### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by mervebdurna**

### CODE REVIEW

File: 1.sql_select.sql [hunk 1]
```diff
@@ -0,0 +1,89 @@
+--case1 :  get all columns
+select * from table_name
+
+--case2 :  gel columns with alias
+select  name as Isim,
```
Comment (durnasaban) [inline 1.sql_select.sql]: should use English words instead of the Turkish words

File: 1.sql_select.sql [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+from table_name
+
+-- case6: get number of distinct value 
+select count(distinct name)
+from table_name
+
+-- case7: recoding 'null' values with COALESCE 
+-- SOURCE : https://towardsdatascience.com/6-sql-tricks-every-data-scientist-should-know-f84be499aea5
+select null_column,
+       COALESCE(null_column, 'Missing') as encoded_null_colum
```
Comment (durnasaban) [inline 1.sql_select.sql]: Column names should be upper camel case

File: 1.sql_select.sql [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from table_name
+
+-- case8: recoding empty and 'NA' values
+-- NOTE :COALESCE don't work with 'NA' and empty instead CASE WHEN statement can be used
+select na_column,
+       CASE WHEN na_column = 'NA' THEN 'NA_Missing' END as encoded_na_column,
```
Comment (durnasaban) [inline 1.sql_select.sql]: Good to have `else` here!

File: 1.sql_select.sql [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+SELECT DATEADD (dd, -DAY (GETDATE () - 1), GETDATE ()) as FirstDayCurrMonth,
+	DATEADD (dd, -DAY (GETDATE ()), GETDATE ()) as LastDayPrevMonth
+
+
+SELECT DATENAME (dw, GETDATE ()) as Weekday,
+	DATENAME (mm, GETDATE ()) as Month
+
+--case 10 :Artihmetic expresion usage
+-- calculate without a table;) 
+Select 2 + 2 as the result
```
Comment (durnasaban) [inline 1.sql_select.sql]: I don't think that query can be compiled successfully!

File: 1.sql_select.sql [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+-- Calculations on the values from the columns 
+-- ROUND function rounds the result to n decimal places
+-- FLOOR (fractional_number) - rounds any floating point number down 
+-- CEILING (fractional_number) works similarly - it always rounds up to the full value
+
+select UnitPrice, Quantity, Discount, 
+	Round ((UnitPrice * Quantity * (1-Discount)) * 1.23, 2) as TotalBrutto
+from dbo. [Order Details]
```
Comment (durnasaban) [inline 1.sql_select.sql]: If the table names don't have a space, that would be better for not making mistakes easily!

File: 1.sql_select.sql [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+SELECT FLOOR (123.78), FLOOR (-123.78), CEILING (123.78), CEILING (-123.78)
```
Comment (durnasaban) [inline 1.sql_select.sql]: Good to have columns names here!

File: 1.sql_select.sql [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+--source : https://www.sqlpedia.pl/funkcje-matematyczne-w-sql/
+
+
+--case 11 :Text expresion usage
+select LastName + '' + FirstName as FirstName,
```
Comment (durnasaban) [inline 1.sql_select.sql]: Should be space between `FirstName` and `LastName`!

File: 2.sql_where.sql [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+where City LIKE '[ab] [np]%'  --Anacortes,Bobingy,Boise,Bonn,Boston,Bothell
+
+-- case 4 : Usage of where instead of join , the following queries are equal.
+-- Outer joins can be specified by using either the (+) syntax in the WHERE clause or the OUTER JOIN keywords in the FROM clause. 
+-- Note : The result of an outer join contains a copy of all rows from one table. In this topic, the table whose rows are preserved is called the “outer” table, and the other table is called the “inner” table.
+-- In a LEFT OUTER JOIN, the left-hand table is the outer table and the right-hand table is the inner table.
+-- In a RIGHT OUTER JOIN, the right-hand table is the outer table and the left-hand table is the inner table.
+
+
+--inner join :
```
Comment (durnasaban) [inline 2.sql_where.sql]: It's not an inner join, it's a self join!

**Review by durnasaban — CHANGES_REQUESTED** [review]

File: 1.sql_select.sql (hunk 1 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

File: 1.sql_select.sql (hunk 2 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

File: 1.sql_select.sql (hunk 3 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

File: 1.sql_select.sql (hunk 4 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

File: 1.sql_select.sql (hunk 6 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done

File: 1.sql_select.sql (hunk 7 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

File: 2.sql_where.sql (hunk 8 above)
Comment (mervebdurna) [reply 2.sql_where.sql]: Thanks for your comment but I don't think it is a self join as the t1 and t2 table are different.

File: 1.sql_select.sql (hunk 5 above)
Comment (mervebdurna) [reply 1.sql_select.sql]: done!!

**Review by mervebdurna — COMMENTED** [review]
Thanks for reviewing, it was really  valuable for me to improve myself:)

File: 2.sql_where.sql (hunk 8 above)
Comment (durnasaban) [reply 2.sql_where.sql]: Please take a look at that url => https://www.w3schools.com/sql/sql_join_self.asp

**Review by durnasaban — CHANGES_REQUESTED** [review]

File: 2.sql_where.sql [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+-- Note : The result of an outer join contains a copy of all rows from one table. In this topic, the table whose rows are preserved is called the “outer” table, and the other table is called the “inner” table.
+-- In a LEFT OUTER JOIN, the left-hand table is the outer table and the right-hand table is the inner table.
+-- In a RIGHT OUTER JOIN, the right-hand table is the outer table and the left-hand table is the inner table.
+
+
+--inner join : 
+select t1.c1, t2.c2
+    from t1, t2
+    where t1.c1 = t2.c2;
+
```
Comment (durnasaban) [inline 2.sql_where.sql:103]: Please add `inner join` description here!

**Review by durnasaban — CHANGES_REQUESTED** [review]

File: 2.sql_where.sql (hunk 8 above)
Comment (mervebdurna) [reply 2.sql_where.sql]: ok:/

File: 2.sql_where.sql (hunk 9 above)
Comment (mervebdurna) [reply 2.sql_where.sql:103]: done

**Review by durnasaban — APPROVED** [review]

**Pull request merged**